src/flamenco/runtime/fd_blockstore.h

```diff
@@ -718,28 +723,50 @@ struct fd_blockstore_ser {
 };
 typedef struct fd_blockstore_ser fd_blockstore_ser_t;
 
-/* Archives a block and block map entry to fd. If fd is -1, no write is attempted */
+/* fd_blockstore_archive_metadata outlines the format of metadata
+   at the start of an archive file - needed so that archive
+   files can be read back on initialization. */
+
+struct fd_blockstore_archive_metadata {
+  ulong fd_size_max;
+  ulong lrw_off;
+  ulong mrw_end;
```

Comment:
add a couple of ulong's for future usage

Reply:
we can just version the file

Reply:
then include a magic number